scripts/common/longhorn.sh

```diff
@@ -52,3 +52,149 @@ function longhorn_install_nfs_utils_if_missing_common() {
+        if [ "$DID_MIGRATE_LONGHORN_PVCS" == "1" ]; then
+            report_addon_start "longhorn-removal" "v1"
+            remove_longhorn
+            report_addon_success "longhorn-removal" "v1"
+        fi
+    fi
+}
+
+# longhorn_pvs_removed returns true when we can't find any pv using the longhorn csi driver.
+function longhorn_pvs_removed() {
+    local pvs
+    pvs=$(kubectl get pv -o=jsonpath='{.items[*].spec.csi.driver}' | grep "longhorn" | wc -l)
+    [ "$pvs" = "0" ]
+}
+
+# remove_longhorn deletes everything longhorn releated: deployments, CR objects, and CRDs.
+function remove_longhorn() {
+    # make sure there aren't any PVs using longhorn before deleting it
+    echo "Waiting for Longhorn PVs to be removed"
+    if ! spinner_until 60 longhorn_pvs_removed; then
+        # sometimes longhorn hangs and we need to restart kubelet to make it work again, we
+        # are going to give this approach a try here before bailing out.
+        printf "${YELLOW}Some Longhorn PVs are still online, trying to restart kubelet.\n${NC}"
+        systemctl restart kubelet
+        echo "Waiting for Longhorn PVs to be removed"
+        if ! spinner_until 60 longhorn_pvs_removed; then
+            printf "%b" "$RED"
+            printf "ERROR: \n"
+            printf "There are still PVs using Longhorn.\n"
+            printf "Remove these PVs before continuing.\n"
+            printf "%b" "$NC"
+            exit 1
+        fi
+    fi
+
+    # scale ekco to 0 replicas if it exists
+    if kubernetes_resource_exists kurl deployment ekc-operator; then
+        kubectl -n kurl scale deploy ekc-operator --replicas=0
+        echo "Waiting for ekco pods to be removed"
+        spinner_until 120 ekco_pods_gone
+    fi
+
+    # remove longhorn volumes first so the operator can correctly delete them.
+    printf "Removing Longhorn volumes:\n"
+    kubectl delete volumes.longhorn.io -n longhorn-system --all
+
+    # once volumes have been gone we can remove all other longhorn CR objects.
+    printf "Removing Longhorn custom resource objects - this may take some time:\n"
+    kubectl get crd | grep 'longhorn' | grep -v 'volumes' | awk '{ print $1 }' | xargs -I'{}' kubectl -n longhorn-system delete '{}' --all
+
+    # delete longhorn CRDs
+    printf "Removing Longhorn custom resources:\n"
+    kubectl get crd | grep 'longhorn' | awk '{ print $1 }' | xargs -I'{}' kubectl delete crd '{}'
+
+    # delete longhorn ns
+    kubectl delete ns longhorn-system
+
+    # delete longhorn storageclass(es)
+    printf "Removing Longhorn StorageClasses"
+    kubectl get storageclass | grep longhorn | awk '{ print $1 }' | xargs -I'{}' kubectl delete storageclass '{}'
+
+    # scale ekco back to 1 replicas if it exists
+    if kubernetes_resource_exists kurl deployment ekc-operator; then
+        kubectl -n kurl scale deploy ekc-operator --replicas=1
+    fi
+
+    # print success message
+    printf "%bRemoved Longhorn successfully!\n%b" "$GREEN" "$NC"
```

Comment:
You could use the wrapper func `logSuccess()` here.

Reply:
fixed.